internal/transform/transform.go

```diff
@@ -106,6 +111,59 @@ func ExtractStyles(doc *astro.Node) {
 	}
 }
 
+func NormalizeSetDirectives(doc *astro.Node) {
+	var nodes []*astro.Node
+	var directives []*astro.Attribute
+	walk(doc, func(n *astro.Node) {
+		if n.Type == astro.ElementNode && HasSetDirective(n) {
+			for _, attr := range n.Attr {
+				if attr.Key == "set:html" || attr.Key == "set:text" {
+					nodes = append(nodes, n)
+					directives = append(directives, &attr)
+					return
+				}
+			}
+		}
+	})
+
+	if len(nodes) > 0 {
+		for i, n := range nodes {
+			directive := directives[i]
+			n.RemoveAttribute(directive.Key)
+			expr := &astro.Node{
+				Type:          astro.ElementNode,
+				Data:          "astro:expression",
+				Expression:    true,
+				RawExpression: true,
+			}
+			loc := make([]loc.Loc, 1)
+			loc = append(loc, directive.ValLoc)
+			data := directive.Val
+			if directive.Key == "set:text" {
+				data = fmt.Sprintf("$$escapeHTML(%s)", data)
+			}
+			expr.AppendChild(&astro.Node{
+				Type: astro.TextNode,
+				Data: data,
+				Loc:  loc,
+			})
+
+			shouldWarn := false
+			// Remove all existing children
+			for c := n.FirstChild; c != nil; c = c.NextSibling {
+				if !shouldWarn {
+					shouldWarn = c.Type == astro.CommentNode || (c.Type == astro.TextNode && len(strings.TrimSpace(c.Data)) != 0)
+				}
+				n.RemoveChild(c)
+			}
```

Comment:
If there are existing children, we remove them. We only throw a warning if the children are element nodes or non-empty text nodes.